pkg/gui/controllers/commit_description_controller.go

```diff
@@ -0,0 +1,50 @@
+package controllers
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/gui/types"
+)
+
+type CommitDescriptionController struct {
+	baseController
+	*controllerCommon
+}
+
+var _ types.IController = &CommitMessageController{}
+
+func NewCommitDescriptionController(
+	common *controllerCommon,
+) *CommitDescriptionController {
+	return &CommitDescriptionController{
+		baseController:   baseController{},
+		controllerCommon: common,
+	}
+}
+
+func (self *CommitDescriptionController) GetKeybindings(opts types.KeybindingsOpts) []*types.Binding {
+	bindings := []*types.Binding{
+		{
+			Key:     opts.GetKey(opts.Config.Universal.TogglePanel),
+			Handler: self.close,
+		},
+		{
+			Key:     opts.GetKey(opts.Config.Universal.Return),
+			Handler: self.close,
+		},
+	}
+
+	return bindings
+}
+
+func (self *CommitDescriptionController) Context() types.Context {
+	return self.context()
+}
+
+// this method is pointless in this context but I'm keeping it consistent
+// with other contexts so that when generics arrive it's easier to refactor
+func (self *CommitDescriptionController) context() types.Context {
+	return self.contexts.CommitMessage
+}
+
+func (self *CommitDescriptionController) close() error {
+	return self.c.PopContext()
```

Comment:
we're in a bit of a pickle here: it should be that when you hit escape on the description panel it just escapes out of the whole thing, but currently it's instead giving focus back to the summary panel. That's because when you press tab in the summary panel it pushes the description panel onto the stack without removing itself from the stack. We want to make it so that when we push the description panel onto the stack, the summary panel is removed beforehand.

In `pkg/gui/context.go` we have some similar logic for this inside `pushToContextStack`. We could update it so that no two persistent popups end can be on the stack at the same time:
```go
			if (topContext.GetKind() == types.TEMPORARY_POPUP && c.GetKey() != context.SEARCH_CONTEXT_KEY) ||
				// we only ever want one main context on the stack at a time.
				(topContext.GetKind() == types.MAIN_CONTEXT && c.GetKind() == types.MAIN_CONTEXT) ||
				// we only ever want one persistent context on the stack at a time.
				(topContext.GetKind() == types.PERSISTENT_POPUP && c.GetKind() == types.PERSISTENT_POPUP) {

				contextsToDeactivate = append(contextsToDeactivate, topContext)
				_, gui.State.ContextManager.ContextStack = slices.Pop(gui.State.ContextManager.ContextStack)
			}
```

This introduces a new problem where our summary stack's on-focus-lost handler is executed which makes it disappear. I'll need to spend some time thinking about how to resolve that conundrum (suggestions welcome!)

Reply:
[In this comment](https://github.com/jesseduffield/lazygit/pull/2390#issuecomment-1404243955) mark2185 mentioned that when toggling the summary and description we should just change focus instead of showing and hiding the description panel. I think that behaviour is better than what is currently implemented.

The ideal behaviour in my opinion would be like this:
- Hit tab and go to the next panel (if on summary go to description and viceversa). Both are visible
- When focused on description, hit escape closes both windows (although I also don't mind just closing description here)
- When focused on summary, hit escape closes both windows

I've made a change locally where upon hitting tab, I  `PushContext()` the description or summary. I'm not sure if this is the best way to toggle between these, because I think the stack will just keep getting bigger and bigger right? 

I wonder if it would make sense to only push a popup to the stack if it's not present there. That way if the stack consists of the commit summary panel and the commit description panel in the latest items we could have a function like `PopContextN` that can remove the last `N` items.